scenarios/kubernetes_e2e.py

```diff
@@ -503,7 +505,10 @@ def main(args):
             attempts_remaining -= 1
             try:
                 # tell kubetest to extract from this location
-                args.kubetest_args.append('--extract=' + read_gcs_path(gcs_path))
+                shared_build_gcs_path = read_gcs_path(gcs_path)
+                args.kubetest_args.append('--extract=' + shared_build_gcs_path)
+                mode.add_environment('SHARED_BUILD_GCS_PATH=' + shared_build_gcs_path)
+                os.environ['SHARED_BUILD_GCS_PATH'] = shared_build_gcs_path
```

Comment:
why not pass this down to kubeadm? Setting global env feels like a bad smell.

Reply:
Do you have a suggestion how to unit test this better then?
I initially just passed an extra arg to `kubeadm_version()`, but that didn't allow faking out the env.

Reply:
Also, if you give me hints how to refactor later, I can do that.
Now I want to see if this works in practice :smile:

Reply:
I think with `--kubeadm=pull` you also want to add `--use-shared-build` in that test, and you can fake an output from `read_gcs_path()`

Reply:
It would definitely be preferable to avoid setting global env, we've already had quite a bit of pain trying to clean these up in general.